### PR TITLE
Vary current and temperature in `"parameter comparison"`

### LINKED ISSUE #57: Vary `"Current function [A]"` and `"Ambient temperature [K]"` in `"parameter comparison"`
You could also vary "Ambient temperature [K]" here as well as the `param_to_vary`. You could also vary "Current function [A]" (keeping it the same for all comparisons). E.g. you recently tweeted "Doyle-Fuller-Newman model with Chen2020 parameters varying 'Positive electrode exchange-current density [A.m-2]' for a 1.0 C discharge at 25.0°C" but you could also do "Doyle-Fuller-Newman model with Chen2020 parameters varying 'Positive electrode exchange-current density [A.m-2]' for a 2.0 C discharge at 35.0°C"

Happy to leave that for a separate PR though.

_Originally posted by @rtimms in https://github.com/pybamm-team/BattBot/pull/48#discussion_r679940314_

As discussed above, it would be nice to vary these parameters in every comparison. Varying both of them in constant discharge `"parameter comparisons"` and just `"Ambient temperature [K]"` with experiments in `"parameter comparisons"`.

### LINKED ISSUE #62: GIF size > 15728640 bytes in some cases
```
2021-08-12 06:54:04,277 - [INFO] upload.post_request(199): 400
2021-08-12 06:54:04,277 - [INFO] upload.post_request(200): {"request":"\/1.1\/media\/upload.json","error":"File size exceeds 15728640 bytes."}
2021-08-12 06:54:04,277 - [INFO] upload.post_request(201): Twitter API internal error. Trying again in 5 minutes
```

[This](https://github.com/pybamm-team/BattBot/pull/61/checks?check_run_id=3308745032) is the first case where I have seen this error.

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Closes #57 
Closes #62 

## Changes / Additions
 - Now that `"Ambient temperature [K]"` is being varied in every comparison and `"Current function [A]"` is being varied in all non-experiment comparisons, it made sense to generate them in `config_generator` rather than in `ComparisonGenerator`.
 - `ComparisonGenerator` now accepts `params` which also removes the need of `reply_overrides` as the values are already varied in `params`.
 - Added required parameters in the `tweet_text` to go with these changes.
 - Removed `"Ambient temperature [K]"` and `"Current function [A]"` from `param_to_vary_dict` as now they are varied in all the cases.

### CODE REVIEW

**Review by rtimms — APPROVED** [review]
looks good, thanks @Saransh-cpp !

File: REQUEST_EXAMPLES.md [hunk 1]
```diff
@@ -16,16 +16,17 @@ The parameter sets or the chemistries that you can use in a tweet (casing of the
...
 ```
 @battbot_ compare SPM and SPMe for a constant dicharge of 1.25C at 290K with Chen2020 parameters #battbot.
 @battbot_ #battbot can you compare Single particle model, Single particle model with electrolyte and DFN model at 300K with a c-rate of 0.5C with Marquis2019 chemistry?
 @battbot_ #battbot compare spm, spme DFN model with a constant dicharge of 0.75C with Ai2020 at 280K
+@battbot #battbot compare spm and spme with chen2020 parameters with the experiment - [("Discharge at C/10 for 10 hours or until 3.3 V","Rest for 1 hour","Charge at 1 A until 4.1 V","Hold at 4.1 V until 50 mA","Rest for 1 hour")] * 3 at 298K
 ```
 
 ### Mandatory keywords -
-Adding ',' anywhere and the casing of the sentence in the tweet text won't effect the simulation.
+Adding ',', ':' or '-' anywhere and the casing of the sentence in the tweet text won't effect the simulation.
```
Comment (rtimms) [inline REQUEST_EXAMPLES.md:29]: Very minor, but I think this should be "affect" and not "effect"

**Comment by rtimms** [comment]
What do you mean by it creates another GIF? As in it does a new simulation in hope of the resulting GIF being smaller?

**Comment by Saransh-cpp** [comment]
Now it resizes the GIF again if the API gives an error

**Review by rtimms — APPROVED** [review]

**Pull request merged**